### PR TITLE
[Merged by Bors] - feat: port Topology.MetricSpace.GromovHausdorffRealized

### PR DESCRIPTION
**Pull request by int-y1**
Proof of `dist_mem_candidates` has been simplified

---

todo:

* [x] `dist_mem_candidates` unsolved goals. maybe an issue with `repeat'`?
* [x] `isCompact_candidatesB` error at `refine' arzela_ascoli₂`

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/MetricSpace/GromovHausdorffRealized.lean [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+    candidatesBOfCandidates f fA ∈ candidatesB X Y :=
+  fA
+#align Gromov_Hausdorff.candidates_b_of_candidates_mem GromovHausdorff.candidatesBOfCandidates_mem
+
+/-- The distance on `X ⊕ Y` is a candidate -/
+private theorem dist_mem_candidates :
+    (fun p : (X ⊕ Y) × (X ⊕ Y) => dist p.1 p.2) ∈ candidates X Y := by
+  simp only [candidates, dist_comm, forall_const, and_true_iff, add_comm, eq_self_iff_true,
+    and_self_iff, Sum.forall, Set.mem_setOf_eq, dist_self]
+  repeat'
```
Comment (apurvnakade) [inline Mathlib/Topology/MetricSpace/GromovHausdorffRealized.lean:235]: I removed the `repeat'` and manually applying the tactics. This is the required order 
```lean
  constructor
  constructor
  constructor
  exact fun x y => by rfl
  exact fun x y => by rfl
  constructor
  exact fun x y => maxVar_bound
```

The error appears in the last step: `  exact fun x y => maxVar_bound`

```lean
typeclass instance problem is stuck, it is often due to metavariables
 Nonempty (?m.99960 x y)
```

I don't know enough about this file to fix this.

Comment (apurvnakade) [reply Mathlib/Topology/MetricSpace/GromovHausdorffRealized.lean:235]: Actually, the last line is supposed to be `  exact fun a y z => dist_triangle_left _ _ _` which throws the error

```lean 
typeclass instance problem is stuck, it is often due to metavariables
 PseudoMetricSpace (?m.99752 a y z)
```

:shrug:

Comment (int-y1) [reply Mathlib/Topology/MetricSpace/GromovHausdorffRealized.lean]: I found a new proof that uses less complicated things

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**